python/tpmath.py

```diff
@@ -75,4 +75,35 @@ def compute_average(source_zones, variables_to_average, constant_variables=None)
             tp.data.operate.execute_equation("{%s} = 0"%(v.name), zones=[avg_zone])
     return avg_zone
 
-
+def compute_phaseAverage(source_zones, zones_per_timeperiod, variables_to_average, constant_variables=None):
+	# currently implemented for constant delta t solution files
+	dataset = source_zones[0].dataset
+	phAvg_zones = dataset.copy_zones(source_zones[slice(zones_per_timeperiod)])
+	# rename zones
+	for zone, i in zip(phAvg_zones, range(zones_per_timeperiod)):
+		zone.name = "%s %.2f pi - %s" % ('phase', 2*i/zones_per_timeperiod, zone.name)
+		zone.strand = dataset.num_zones
+	# all_zNames = dataset.zone_names
```

Comment:
Here I tried to update the names without looping, but the names would not update.
These commented lines when run from outside the function works ...
but inside the function do not update the names. so temporarily i used the for loop

Reply:
I see it now. `Dataset.zone_names` just returns a list of strings that are no longer tied to a objects in the Tecplot Engine. The loop is the way to go here.